crates/net/network/tests/it/connect.rs

```diff
@@ -582,6 +582,47 @@ async fn test_shutdown() {
     assert_eq!(handle0.num_connected_peers(), 0);
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn test_exceed_outgoing_connections() {
+    let net = Testnet::create(2).await;
+    let secret_key = SecretKey::new(&mut rand::thread_rng());
+    let peers_config = PeersConfig::default().with_max_outbound(1);
+
+    let config = NetworkConfigBuilder::<EthNetworkPrimitives>::new(secret_key)
```

Comment:
would you like to introduce a new fn for this `fn eth`, similar to

https://github.com/paradigmxyz/reth/blob/5cf0276f1e252d8d16e5fd76ba82776fff0e99db/crates/net/network/src/manager.rs#L154-L154

?

Reply:
sure, will do 🫡